README.rst

```diff
@@ -9,37 +9,154 @@ Kinto Distribution
+which is default) in one terminal first:
+
+.. code-block:: shell
+
+    $ docker-compose up web
+
+Now, in a separate terminal, first check that you can reach the Kinto
+server:
+
+.. code-block:: shell
+
+    $ curl http://localhost:8888/v1/__heartbeat__
+    $ docker-compose run tests
+
+Debugging Locally (advanced)
+----------------------------
+
+Suppose you want to play with running the Kinto server, then go into
+a ``bash`` session like this:
+
+.. code-block:: shell
+
+    $ docker-compose run --service-ports --user 0 web bash
+
+Now you're ``root`` so you can do things like ``apt-get update && apt-get install jed``
+to install tools and editors. Also, because of the ``--service-ports`` if you do
+start a Kinto server on ``:8888`` it will be exposed from the host.
+
+For example, instead of starting Kinto with ``uwsgi`` you can start it
+manually with ``kinto start``:
+
+.. code-block:: shell
+
+    $ kinto start --ini config/example.ini
+
+Another thing you might want to debug is the ``tests`` container that does
+the ``curl`` commands against the Kinto server. But before you do that,
+you probably want to start the services:
+
+.. code-block:: shell
+
+    $ docker-compose up web
+
+.. code-block:: shell
+
+    $ docker-compose run tests bash
+
+Now, from that ``bash`` session you can reach the other services like:
+
+.. code-block:: shell
+
+    $ curl http://autograph:8000/__heartbeat__
+    $ curl http://web:8000/v1/__heartbeat__
+
+
+Upgrade Things
+--------------
+
+Most common use-case with ``kinto-dist`` is that you want to upgrade one
+of the dependencies. All dependencies are listed in:
+
+* ``requirements/default.txt``
+* ``requirements/constraints.txt``
+* ``requirements/prod.txt``
+
+If there's a package you want to upgrade or add, do that to the
+``requirements/default.txt`` file. If you find out that what you're adding
+requires its own dependencies, add that to ``requirements/constraints.txt``.
+
+To upgrade the requirements file, install `hashin <https://pypi.org/project/hashin/>`_
+globally on your laptop and then run the following (example) command:
+
+.. code-block:: shell
+
+    $ hashin -r requirements/default.txt myhotnewpackage
+
+Or if you know the exact version you need:
+
+.. code-block:: shell
+
+    $ hashin -r requirements/default.txt myhotnewpackage==1.2.3
+
+If you just want to *upgrade* an existing package, based on the latest version
+available on `PyPi <https://pypi.org/>`_ you do it as if it's a new package.
+For example:
+
+.. code-block:: shell
+
+    $ hashin -r requirements/default.txt requests
+
+To test that this installs run:
+
+.. code-block:: shell
+
+    $ docker-compose build web
+
+If it fails because ``pip`` believes your new package has other dependencies
+not already mentioned in ``requirements/constraints.txt`` add them like this:
+
+.. code-block:: shell
+
+    $ hashin -r requirements/constraints.txt imneedy alsoneeded
```

Comment:
Will Renovate take care of all of this?

Reply:
Not for new packages but we should perhaps mention that Renovate can take care of upgrading existing packages. Granted, we don't have Renovate yet.